motionplan/motionPlanner.go

```diff
@@ -186,6 +188,34 @@ func Replan(ctx context.Context, request *PlanRequest, currentPlan Plan, replanC
 	if err != nil {
 		return nil, err
 	}
+	// Check if the PlanRequest's options specify "complex"
+	// This is a list of poses in the same frame as the goal through which the robot must pass
+	if complexOption, ok := request.Options["complex"]; ok {
+		if complexPosesList, ok := complexOption.([]interface{}); ok {
+			// If "complex" is specified and is a list of PoseInFrame, use it for planning.
+			requestCopy := *request
+			delete(requestCopy.Options, "complex")
```

Comment:
Why make a copy and delete this key?

Reply:
To avoid altering the original object, which may be otherwise in use elsewhere.